src/rust-crypto/verification.ts

```diff
@@ -0,0 +1,429 @@
+    public get otherUserId(): string {
+        return this.inner.otherUserId.toString();
+    }
+
+    /** For verifications via to-device messages: the ID of the other device. Otherwise, undefined. */
+    public get otherDeviceId(): string | undefined {
+        return this.inner.otherDeviceId?.toString();
+    }
+
+    /** True if the other party in this request is one of this user's own devices. */
+    public get isSelfVerification(): boolean {
+        return this.inner.isSelfVerification();
+    }
+
+    /** current phase of the request. */
+    public get phase(): VerificationPhase {
+        const phase = this.inner.phase();
+
+        switch (phase) {
+            case RustSdkCryptoJs.VerificationRequestPhase.Created:
+            case RustSdkCryptoJs.VerificationRequestPhase.Requested:
+                return VerificationPhase.Requested;
+            case RustSdkCryptoJs.VerificationRequestPhase.Ready:
+                return VerificationPhase.Ready;
+            case RustSdkCryptoJs.VerificationRequestPhase.Transitioned:
+                return VerificationPhase.Started;
+            case RustSdkCryptoJs.VerificationRequestPhase.Done:
+                return VerificationPhase.Done;
+            case RustSdkCryptoJs.VerificationRequestPhase.Cancelled:
+                return VerificationPhase.Cancelled;
+        }
+
+        throw new Error(`Unknown verification phase ${phase}`);
+    }
+
+    /** True if the request has sent its initial event and needs more events to complete
+     * (ie it is in phase `Requested`, `Ready` or `Started`).
+     */
+    public get pending(): boolean {
+        throw new Error("not implemented");
+    }
+
+    /**
+     * True if we have started the process of sending an `m.key.verification.ready` (but have not necessarily received
+     * the remote echo which causes a transition to {@link VerificationPhase.Ready}.
+     */
+    public get accepting(): boolean {
+        throw new Error("not implemented");
+    }
+
+    /**
+     * True if we have started the process of sending an `m.key.verification.cancel` (but have not necessarily received
+     * the remote echo which causes a transition to {@link VerificationPhase.Cancelled}).
+     */
+    public get declining(): boolean {
+        throw new Error("not implemented");
+    }
+
+    /**
+     * The remaining number of ms before the request will be automatically cancelled.
+     *
+     * `null` indicates that there is no timeout
+     */
+    public get timeout(): number | null {
+        throw new Error("not implemented");
+    }
+
+    /** once the phase is Started (and !initiatedByMe) or Ready: common methods supported by both sides */
+    public get methods(): string[] {
+        throw new Error("not implemented");
+    }
+
+    /** the method picked in the .start event */
+    public get chosenMethod(): string | null {
+        const verification: RustSdkCryptoJs.Qr | RustSdkCryptoJs.Sas | undefined = this.inner.getVerification();
```

Comment:
The `RustSdkCryptos.Qr | RustSdkCryptoJs.Sas | undefined` union appears a couple of times in this class. Do you think it would warrant becoming a type?